core/templates/domain/skill/editable-skill-backend-api.service.spec.ts

```diff
@@ -15,67 +15,58 @@
 /**
  * @fileoverview Unit tests for EditableSkillBackendApiService.
  */
+import { HttpClientTestingModule, HttpTestingController } from
+  '@angular/common/http/testing';
+import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';
 
-require('domain/editor/undo_redo/undo-redo.service.ts');
-require('domain/skill/editable-skill-backend-api.service.ts');
-require('services/csrf-token.service.ts');
-// TODO(#7222): Remove the following block of unnnecessary imports once
-// the code corresponding to the spec is upgraded to Angular 8.
-import { UpgradedServices } from 'services/UpgradedServices';
-// ^^^ This block is to be removed.
-
+import { CsrfTokenService } from
+  'services/csrf-token.service';
+import { EditableSkillBackendApiService } from
+  'domain/skill/editable-skill-backend-api.service';
 
 describe('Editable skill backend API service', function() {
-  var EditableSkillBackendApiService = null;
-  var UndoRedoService = null;
-  var $httpBackend = null;
-  var sampleResponse = null;
-  var CsrfService = null;
-  var sampleResponse2 = null;
-
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
-    }
-  }));
-
-  beforeEach(angular.mock.inject(function($injector, $q) {
-    EditableSkillBackendApiService = $injector.get(
-      'EditableSkillBackendApiService');
-    UndoRedoService = $injector.get('UndoRedoService');
-    $httpBackend = $injector.get('$httpBackend');
-    CsrfService = $injector.get('CsrfTokenService');
-
-    spyOn(CsrfService, 'getTokenAsync').and.callFake(function() {
-      var deferred = $q.defer();
-      deferred.resolve('sample-csrf-token');
-      return deferred.promise;
+  let editableSkillBackendApiService = null;
+  let httpTestingController: HttpTestingController;
+  let sampleResponse = null;
+  let csrfService = null;
+  let sampleResponse2 = null;
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
     });
+    editableSkillBackendApiService =
+      TestBed.get(EditableSkillBackendApiService);
+    httpTestingController = TestBed.get(HttpTestingController);
+    csrfService = TestBed.get(CsrfTokenService);
+    spyOn(csrfService, 'getTokenAsync').and.callFake(() => {
+      return new Promise((resolve) => {
+        resolve('sample-csrf-token');
+      });
+    }
+    );
 
-    var misconceptionDict1 = {
+    let misconceptionDict1 = {
```

Comment:
Can't the following variables be declared as const since they are not being re-assigned?

Reply:
Done!